frontend/riskofbias/robStudyForm/store.js

```diff
@@ -41,54 +23,109 @@ class RobFormStore {
+    }
+
+    @computed get metricDomains() {
+        let {domains} = this;
+        return _.chain(this.settings.metrics)
+            .map(metric => [metric.id, domains[metric.domain_id]])
+            .fromPairs()
+            .value();
+    }
+
+    updateRobScore(score, riskofbias, overrideOptions) {
+        let overrideDataTypeValue;
+        if (score.is_default) {
+            overrideDataTypeValue = null;
+        } else if (score.overridden_objects.length > 0) {
+            overrideDataTypeValue = score.overridden_objects[0].content_type_name;
+        } else {
+            overrideDataTypeValue = _.keys(overrideOptions)[0] || null;
+        }
+        return Object.assign({}, score, {
+            overrideDataTypeValue,
+            author: riskofbias.author,
+            final: riskofbias.final,
+            score_symbol: this.settings.score_metadata.symbols[score.score],
+            score_shade: this.settings.score_metadata.colors[score.score],
+            score_description: this.settings.score_metadata.choices[score.score],
+        });
+    }
+
+    getMetricIds(domainId) {
+        return _.uniq(
+            store.editableScores.reduce(function(array, score) {
+                store.metricDomains[score.metric_id].id == domainId
+                    ? array.push(score.metric_id)
+                    : null;
+                return array;
+            }, [])
+        );
     }
+
     getEditableScoresForMetric(metricId) {
-        return [...this.editableScores.values()].filter(score => score.metric.id == metricId);
+        return this.editableScores.filter(score => score.metric_id == metricId);
     }
     getNonEditableScoresForMetric(metricId) {
-        return this.nonEditableScores.filter(score => score.metric.id == metricId);
+        return this.nonEditableScores.filter(score => score.metric_id == metricId);
     }
-    metricHasOverrides(metricId) {
-        return _.chain(this.scores)
-            .filter(score => score.metric.id == metricId)
+
+    nonEditableMetricHasOverrides(metricId) {
+        return _.chain(this.getNonEditableScoresForMetric(metricId))
             .map(score => score.is_default === false)
             .some()
             .value();
     }
-    editableMetricHasOverride(metricId) {
+    editableMetricHasOverrides(metricId) {
         return _.chain(this.getEditableScoresForMetric(metricId))
             .map(score => score.is_default === false)
             .some()
             .value();
     }
-    getEditableScore(scoreId) {
-        if (!this.editableScores.has(scoreId)) {
-            throw `Score ${scoreId} does not exist.`;
-        }
-        return this.editableScores.get(scoreId);
-    }
 
     // actions
     @action.bound setConfig(elementId) {
         this.config = JSON.parse(document.getElementById(elementId).textContent);
     }
 
+    @action.bound fetchSettings(assessment_id) {
+        const url = robSettingsUrl(assessment_id);
```

Comment:
idea - we're using this same pattern in all the stores; what we changed the constants to do the fetch?

```javascript
const robSettingsUrl = function(assessment_id, callback){
     fetch(url, h.fetchGet)
            .then(response => response.json())
            .then(data => callback(data));
}
```